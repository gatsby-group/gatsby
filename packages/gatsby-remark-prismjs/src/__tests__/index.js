const remark = require(`remark`)
const plugin = require(`../index`)

describe(`remark prism plugin`, () => {
  it(`generates a <pre> tag with a language class by default`, () => {
    const code = '```js\n// Fake\n```'
    const markdownAST = remark.parse(code)
    plugin({ markdownAST })
    expect(markdownAST).toMatchSnapshot()
  })

  it(`generates a <pre> tag with a data attribute if configured to do so`, () => {
    const code = '```js\n// Fake\n```'
    const markdownAST = remark.parse(code)
    plugin({ markdownAST }, { useDataAttribute: true})
    expect(markdownAST).toMatchSnapshot()
  })
})
