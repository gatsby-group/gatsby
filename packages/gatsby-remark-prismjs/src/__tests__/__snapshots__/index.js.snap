// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`remark prism plugin generates a <pre> tag with a data attribute if configured to do so 1`] = `
Object {
  "children": Array [
    Object {
      "lang": "js",
      "position": Position {
        "end": Object {
          "column": 4,
          "line": 3,
          "offset": 17,
        },
        "indent": Array [
          1,
          1,
        ],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "html",
      "value": "<div class=\\"gatsby-highlight\\">
      <pre data-language=\\"js\\"><code><span class=\\"token comment\\" spellcheck=\\"true\\">// Fake</span>
</code></pre>
      </div>",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 4,
      "line": 3,
      "offset": 17,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`remark prism plugin generates a <pre> tag with a language class by default 1`] = `
Object {
  "children": Array [
    Object {
      "lang": "js",
      "position": Position {
        "end": Object {
          "column": 4,
          "line": 3,
          "offset": 17,
        },
        "indent": Array [
          1,
          1,
        ],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "html",
      "value": "<div class=\\"gatsby-highlight\\">
      <pre class=\\"language-js\\"><code><span class=\\"token comment\\" spellcheck=\\"true\\">// Fake</span>
</code></pre>
      </div>",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 4,
      "line": 3,
      "offset": 17,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;
